internal/ledger/compare.go

```diff
@@ -0,0 +1,413 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ledger
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric/common/ledger/util"
+	"github.com/hyperledger/fabric/core/ledger/kvledger"
+	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/privacyenabledstate"
+	"github.com/hyperledger/fabric/internal/fileutil"
+	"github.com/pkg/errors"
+)
+
+// nsKey type is used to compare between snapshot records using both the namespace and key
+type nsKey struct {
+	namespace string
+	key       []byte
+}
+
+// DiffRecord type for representing diverging records in json
```

Comment:
Just a nit-picking: Here and other place - usually, golang comments sound like "DiffRecord represents a diverging record in json"